base/uk.ac.stfc.isis.ibex.ui.graphing/src/uk/ac/stfc/isis/ibex/ui/graphing/websocketview/MatplotlibFigure.java

```diff
@@ -154,10 +179,23 @@ public void mouseUp(MouseEvent e) {
 					    plotMessage.setText(e.getNewValue().toString());
 					}
 				});
-		imageListener = viewModel.getImage()
-				.addUiThreadPropertyChangeListener(e -> drawImage((ImageData) e.getNewValue()));
-		
-		
+		cursorTypeListener = viewModel.getCursorType()
+				.addUiThreadPropertyChangeListener(e -> {
+					if (!plotMessage.isDisposed()) {
```

Comment:
Slightly confused about why this refers to the plot message being disposed - maybe you meant the container being disposed - that would make more sense?